scripts/smeshr/MODIS-STC/MODIS_albedo_basin.sh

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env bash
+
+# Script to extract albedo values for a model domain from MODIS.
+# This prepares a NetCDF file with per grid cell albedo values for every
+# hour of one day. It can be directly used to calculate net solar 
+# with incoming shortwave radiation.
+
+# Arguments:
+# ${1}: /path/to/MODIS/inputs
+# ${2}: water year
+
+set -e
+
+extract_modis() {
+  # command from UofU snow-rs package
+  # https://github.com/UofU-Cryosphere/snow-rs
+  # Converts the MODIS matlab to GeoTiff files
+  variable_from_modis --source-dir ${1} \
+                      --water-year ${2} \
+                      --variable albedo_observed_muZ
+
+  if [ $? != 0 ]; then
+    echo "Error extracting variable from MODIS inputs"
+    exit 1
+  fi
+}
+
+# Basin name and processing parameters
+#export BASIN='yampa'
+#BASIN_EXTENT='156726.4 4426192.0 365326.4 4530492.0' #yampa
+export BASIN='littleanimas_jmh'
+BASIN_EXTENT='251200 4170400 256400 4177500'
+RES=100
+EPSG='EPSG:32613'
+BASIN_DOMAIN=" -t_srs $EPSG -tr $RES $RES -dstnodata 65535 -te $BASIN_EXTENT"
+
+SLURM_NTASKS=4
+
+export BASIN_NC="_${BASIN}.nc"
+export BASIN_ONE_DAY_SUFFIX="_24.nc"
+
+export CDO_call="cdo -O -z zip_4 -f nc4 -s"
+export PARALLEL_call="parallel --tagstring {#} --tag --line-buffer --jobs ${SLURM_NTASKS}"
+
+# Extracts the BASIN domain from the Western US MODIS albedo GeoTiffs
+modis_basin() {
+  BASIN_TMP_CUBIC=${1/\.tif/_cubic_tmp.vrt}
+  BASIN_TMP_NN=${1/\.tif/_nn_tmp.vrt}
+  BASIN_TMP=${1/\.tif/_tmp.tif}
+  BASIN_TMP_NC=${1/\.tif/_tmp.nc}
+  BASIN_DOMAIN="${@:2}"
+  # add catch for existing 24.nc files (final output), remove later as needed
+  if [[ -f "${1%.*}_${BASIN}${BASIN_ONE_DAY_SUFFIX}" ]]; then
+    return
+  else
+    echo "${1%.*}_${BASIN}${BASIN_ONE_DAY_SUFFIX}" DNE
+    # echo "BASIN_DOMAIN: ${BASIN_DOMAIN}"
+    FILTER_MATH="A*(A<=numpy.max(B)) + numpy.max(B)*(A>numpy.max(B))" 
+    echo "gdalwarp -q -overwrite -multi \
+      -r cubic ${BASIN_DOMAIN} \
+      ${1} ${BASIN_TMP_CUBIC}"
+
+    # Use cubic resampling to smooth the image
+    gdalwarp -q -overwrite -multi \
+      -r cubic ${BASIN_DOMAIN} \
+      ${1} ${BASIN_TMP_CUBIC}
+
+    # Generate a nearest neighbor to get the max basin value as filter
+    gdalwarp -q -overwrite -multi \
+      ${BASIN_DOMAIN} \
+      ${1} ${BASIN_TMP_NN}
+
+    # Combine products, filter any values from cubic resampling higher
+    # than the max of nearest neighbor and set to that value
+    gdal_calc.py --overwrite --quiet --co COMPRESS=LZW \
+      -A ${BASIN_TMP_CUBIC} -B ${BASIN_TMP_NN} \
+      --calc="${FILTER_MATH}" \
+      --outfile=${BASIN_TMP}
+
+    # Add threshold enforcement of the values lower than the
+    # minimum of the nearest neighbor resampling and set to that min
+    gdal_calc.py --overwrite --quiet --co COMPRESS=LZW \
+      -A ${BASIN_TMP_CUBIC} -B ${BASIN_TMP_NN} \
+      --calc="A*(A>=numpy.min(B)) + numpy.min(B)*(A<numpy.min(B))" \
+      --outfile=${BASIN_TMP}
```

Comment:
Thinking this out loud, I think this step now just voided the max filter enforcement by overwriting the `${BASIN_TMP}`.
From my perspective, this needs to create a third temporary file that is based of the max enforcement.

Pseudo code:
```shell
# Around line ~49
BASIN_TMP_MAX=${1/\.tif/_tmp_max.tif}
BASIN_TMP=${1/\.tif/_tmp.tif}

# Updated max call
gdal_calc.py --overwrite --quiet --co COMPRESS=LZW \
  -A ${BASIN_TMP_CUBIC} -B ${BASIN_TMP_NN} \
  --calc="${FILTER_MATH}" \
  --outfile=${BASIN_TMP_MAX}

# Updated min call
gdal_calc.py --overwrite --quiet --co COMPRESS=LZW \
  -A ${BASIN_TMP_MAX} -B ${BASIN_TMP_NN} \
  --calc="A*(A>=numpy.min(B)) + numpy.min(B)*(A<numpy.min(B))" \
  --outfile=${BASIN_TMP}
```
```

Reply:
Updated in [latest commit](https://github.com/UofU-Cryosphere/isnoda/pull/64/commits/97250ea6315b0e2cf46c03974824d17ea8337f76)